crates/autopilot/src/solvable_orders.rs

```diff
@@ -539,10 +547,15 @@ fn orders_with_balance(
     mut orders: Vec<Order>,
     balances: &Balances,
     settlement_contract: H160,
+    disable_1271_order_balance_filter: bool,
 ) -> Vec<Order> {
     // Prefer newer orders over older ones.
     orders.sort_by_key(|order| std::cmp::Reverse(order.metadata.creation_date));
     orders.retain(|order| {
+        if disable_1271_order_balance_filter && matches!(order.signature, Signature::Eip1271(_)) {
+            return true;
+        }
+
```

Comment:
Seeing the code block below really makes me think we should detect this based on flashloan hints in app data rather than these flags. It looks like full_app_data is available on the order and there is already a `Validator` component used for adding the partner fee logic that is able to parse the app data in the autopilot.